README.md

```diff
@@ -10,10 +10,438 @@ gem 'registers-ruby-client', git: 'https://github.com/openregister/registers-rub
+entry.timestamp
+
+```
+</details>
+<details>
+<summary>
+Expected output (click here to expand):
+ </summary>
+
+```
+2016-04-05T13:23:05Z
+```
+</details>
+
+#### `item_hash`
+
+Gets the SHA-256 hash of the item which the entry points to.
+
+<details>
+<summary>
+Example usage (click here to expand):
+</summary>
+
+```
+
+register_data = registers_client.get_register 'country', 'beta'
+
+entry = register_data.get_entries.select {|entry| entry.key == 'CZ'}.first
+entry.item_hash
+
+```
+</details>
+<details>
+<summary>
+Expected output (click here to expand):
+ </summary>
+
+```
+sha-256:c45bd0b4785680534e07c627a5eea0d2f065f0a4184a02ba2c1e643672c3f2ed
+```
+</details>
+
+#### `value`
+
+Returns the entry as a hash.
+
+<details>
+<summary>
+Example usage (click here to expand):
+</summary>
+
+```
+
+register_data = registers_client.get_register 'country', 'beta'
+
+entry = register_data.get_entries.select {|entry| entry.key == 'CZ'}.first
+entry.value.to_json
+
+```
+</details>
+<details>
+<summary>
+Expected output (click here to expand):
+ </summary>
+
+```
+"{"key":"CZ","timestamp":"2016-04-05T13:23:05Z","item_hash":"sha-256:c45bd0b4785680534e07c627a5eea0d2f065f0a4184a02ba2c1e643672c3f2ed"}"
+```
+</details>
+
+### Record
+
+#### `entry`
+
+Gets the `Entry` object associated with the record.
+
+<details>
+<summary>
+Example usage (click here to expand):
+</summary>
+
+```
+
+register_data = registers_client.get_register 'country', 'beta'
+
+record = register_data.get_records.select {|record| record.entry.key == 'CZ'}.first
+record.entry.to_json
+
+```
+</details>
+<details>
+<summary>
+Expected output (click here to expand):
+ </summary>
+
+```
+"{"entry_number":205,"parsed_entry":{"key":"CZ","timestamp":"2016-11-11T16:25:07Z","item_hash":"sha-256:c69c04fff98c59aabd739d43018e87a25fd51a00c37d100721cc68fa9003a720"}}"
+```
+</details>
+
+#### `item`
+
+Gets the `Item` object associated with the record.
+
+<details>
+<summary>
+Example usage (click here to expand):
+</summary>
+
+```
+
+register_data = registers_client.get_register 'country', 'beta'
+
+record = register_data.get_records.select {|record| record.entry.key == 'CZ'}.first
+record.item.to_json
+
+```
+</details>
+<details>
+<summary>
+Expected output (click here to expand):
+ </summary>
+
+```
+"{"item_json":"{\"citizen-names\":\"Czech\",\"country\":\"CZ\",\"name\":\"Czechia\",\"official-name\":\"The Czech Republic\",\"start-date\":\"1993-01-01\"}","item_hash":"sha-256:c69c04fff98c59aabd739d43018e87a25fd51a00c37d100721cc68fa9003a720","parsed_item":null}"
+```
+</details>
+
+## Collections
+
+The majority of the methods available in the `RegisterClient` (described in more detail in the next section) return one of three types of collection object. These collections all include `Enumerable` and implement the `each` method.
+
+### EntryCollection
+
+A collection of `Entry` objects.
+
+#### `each`
+
+Yields each `Entry` object in the collection.
```

Comment:
This is because the `each` method _is_ technically `yield`ing the results, but we could change this to `gets` instead.

Reply:
The reason I asked was because `gets` was used earlier in the same doc, but it strikes me that `yields` seems more technically appropriate; `gets` would seem to conjure up the idea of `GET`. As long as we can be consistent and use the same word across the doc, it doesn't matter.

Reply:
We can use `yield` where it's appropriate in the definition of the `each` method, but this doesn't happen elsewhere so we can't be consistent in that sense. How would you like to proceed on this?

Reply:
I see what you mean. In that case I would suggest that we leave it for now and I'll make this another one to review in the context of the whole document.